lib/valueflow.cpp

```diff
@@ -792,7 +792,7 @@ static void setTokenValue(Token* tok, const ValueFlow::Value &value, const Setti
                     if (value1.tokvalue->tokType() == Token::eString) {
                         const std::string s = value1.tokvalue->strValue();
                         const MathLib::bigint index = value2.intvalue;
-                        if (index == s.size()) {
+                        if (static_cast<std::string::size_type>(index) == s.size()) {
```

Comment:
In my opinion it was a mistake to make std::string::size() return value unsigned. I'd rather use the ssize() that will be available in c++20.. maybe introduce a ssize() function in utils.h will work for now.

Reply:
This will be done implicitly anyways - I just doing it explicitly so UBSAN know it is actually intended.

Reply:
I know what you are saying but I dislike putting casts in the code. The code is changed. Imagine that we someday change bigint so it's a `int128_t`. Now your cast will cause loss of precision. The `s.size()` should be promoted.